### PR TITLE
Sanitizer example

### PR DESCRIPTION
**Pull request by jfgoog**
Sanitizer example with build variants for asan and hwasan.

### CODE REVIEW

**Comment by jfgoog** [comment]
Still TODO:
* README.md
* .google/packaging.yaml

File: sanitizers/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            version '3.18.1'
+        }
+    }
+    buildFeatures {
+        viewBinding true
+    }
+}
+androidComponents {
+    onVariants(selector().withBuildType("asan"), {
+        packaging.dex.useLegacyPackaging.set(true)
```
Comment (DanAlbert) [inline sanitizers/app/build.gradle:63]: Is there a bug that needs to be filed?

File: sanitizers/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+        hwasan {
```
Comment (DanAlbert) [inline sanitizers/app/build.gradle:26]: Is it possible to use that build.gradle defined virtual device thing to automatically run this on a compatible system image?

File: sanitizers/app/src/asan/resources/lib/arm64-v8a/wrap.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/system/bin/sh
+# Copied from https://android.googlesource.com/platform/ndk/+/refs/heads/master/wrap.sh/asan.sh
```
Comment (DanAlbert) [inline sanitizers/app/src/asan/resources/lib/arm64-v8a/wrap.sh:2]: Bonus points if you work out the gradle-fu to copy this from the NDK so it doesn't get stale (every so often we need to extend the jdwp workarounds to account for an ART change)

File: sanitizers/app/src/main/cpp/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        SHARED
+
+        # Provides a relative path to your source file(s).
+        native-lib.cpp)
+
+if(SANITIZE)
+if(${SANITIZE} STREQUAL "hwasan")
+    message("Using hwasan")
+    target_compile_options(sanitizers PUBLIC -fsanitize=hwaddress -fno-omit-frame-pointer)
+    set_target_properties(sanitizers PROPERTIES LINK_FLAGS -fsanitize=hwaddress)
```
Comment (DanAlbert) [inline sanitizers/app/src/main/cpp/CMakeLists.txt:30]: You should be able to use https://cmake.org/cmake/help/latest/command/target_link_options.html these days. We have 3.22 available.

**Review by DanAlbert — APPROVED** [review]

File: sanitizers/app/src/main/cpp/CMakeLists.txt (hunk 4 above)
Comment (jfgoog) [reply sanitizers/app/src/main/cpp/CMakeLists.txt]: Done.

File: sanitizers/app/src/asan/resources/lib/arm64-v8a/wrap.sh (hunk 3 above)
Comment (jfgoog) [reply sanitizers/app/src/asan/resources/lib/arm64-v8a/wrap.sh]: Done.

File: sanitizers/app/build.gradle (hunk 2 above)
Comment (jfgoog) [reply sanitizers/app/build.gradle:26]: Maybe. x86_64 and aarch64 both support hwasan. For now, I think I'm just going to point to the hwasan docs in developer.android.com.

**Pull request merged**